### PR TITLE
evaluation review

### PR DESCRIPTION
**Pull request by yeonir**
## 📑 Description
I have added the new prompts for evaluation dimension under a new class in the evaluators.py file

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed
- [x] Branch name follows `type/descript` (e.g. `feature/add-llm-agents`)
- [x] Ready for code review

### CODE REVIEW

File: examples/experiment_eval.py [hunk 1]
```diff
@@ -147,7 +148,7 @@ def _iterate_env_agent_combo_not_in_db(
                 terminal_evaluators=[
                     ReachGoalLLMEvaluator(
                         model_names["env"],
-                        EvaluationForTwoAgents[SotopiaDimensions],
+                        EvaluationForTwoAgents[EvaluationByGroup],
```
Comment (XuhuiZhou) [inline examples/experiment_eval.py:151]: This `EvaluationByGroup` should only be applied once? @yeonir @bugsz Any chance we can do a class like `EvaluationForGroup` to replace the `EvaluationForTwoAgents`?

Comment (XuhuiZhou) [inline examples/experiment_eval.py:151]: Please don't just directly replace the `SotopiaDimensions`, make this a hyperparameter? And the default is to use `SotopiaDimensions`

File: sotopia/envs/evaluators.py [hunk 2]
```diff
@@ -150,6 +150,98 @@ def minus_ten_to_zero_validator(cls, v: tuple[str, int]) -> tuple[str, int]:
...
+class EvaluationByGroup(BaseModel):
+    effectiveness: tuple[str, int] = Field(
+        ...,
+        description="""Consider features of verbosity (e.g. word count, sentence lengths, readability)
+         to analyze the interaction episode between the agents and use this as a method in evaluating 
+         the efficiency of this episode. Higher levels of efficiency should be reflected with focused, 
+         coherent, and collaborative communication that advances the team's objectives. Lower efficiency 
+         can be marked by disjointed, repetitive, or ineffective exchanges that hinder or delay the progress
+         in attaining the goal."""
+        """Output your reasoning process to the 'reasoning' field. Output an integer score ranging from 0 and
```
Comment (XuhuiZhou) [inline sotopia/envs/evaluators.py:162]: The format here is a little bit ambiguous, please refer to the latest prompt to make things clear: https://github.com/sotopia-lab/sotopia/blob/1939bc6963f3c317a9bf62600126d24965f9969c/sotopia/envs/evaluators.py#L156

File: sotopia/envs/evaluators.py [hunk 3]
```diff
@@ -150,6 +150,98 @@ def minus_ten_to_zero_validator(cls, v: tuple[str, int]) -> tuple[str, int]:
...
+        10 in the 'score' field. A higher score indicates a higher level of transactivity.""",
+    )
+
+    @validator("effectiveness", "equity", "transactivity")
+    def zero_to_ten_validator(cls, v: tuple[str, int]) -> tuple[str, int]:
+        assert v[1] >= 0 and v[1] <= 10
+        return v
+    
+
+class EvaluationByGroupPlus(BaseModel):
```
Comment (XuhuiZhou) [inline sotopia/envs/evaluators.py:193]: # TODO: 
check if this is working better than the `EvaluationByGroup`, if not, consider removing it.

**Review by XuhuiZhou — CHANGES_REQUESTED** [review]
Kudos to the first pull request!!! Great job.

**Comment by XuhuiZhou** [comment]
@yeonir Please run: 

`pre-commit run --all-files` to make sure the code format is correct

File: examples/experiment_eval.py (hunk 1 above)
Comment (bugsz) [reply examples/experiment_eval.py:151]: I think in this case we should use `TeamEvaluationDimensions` (something like this) as the name?
`EvaluationForGroup` needs include evaluations for multiple (>=3) agents?

Comment (bugsz) [reply examples/experiment_eval.py:151]: Agree. That needs to be specified in config file

File: examples/experiment_eval.py [hunk 4]
```diff
@@ -148,7 +150,7 @@ def _iterate_env_agent_combo_not_in_db(
                 terminal_evaluators=[
                     ReachGoalLLMEvaluator(
                         model_names["env"],
-                        EvaluationForTwoAgents[EvaluationByGroup],
+                        EvaluationForTwoAgents[evaluation_class],  # type: ignore
```
[the remark this answers is not in the document]
Comment (ProKil) [reply examples/experiment_eval.py:153]: Can you change it to # type: ignore[<your specific ignored type?>]

**Comment by ProKil** [comment]
This issue is stale for too long. Feel free to reopen if it is improved.

**Pull request closed without merging**